yt/frontends/amrvac/data_structures.py

```diff
@@ -227,27 +227,30 @@ def __init__(
     @classmethod
     def _is_valid(cls, filename, *args, **kwargs):
         """At load time, check whether data is recognized as AMRVAC formatted."""
-        validation = False
-        if filename.endswith(".dat"):
-            try:
-                with open(filename, mode="rb") as istream:
-                    fmt = "=i"
-                    [datfile_version] = struct.unpack(
+        if not filename.endswith(".dat"):
+            return False
+        try:
+            with open(filename, mode="rb") as istream:
+                fmt = "=i"
+                [datfile_version] = struct.unpack(
+                    fmt, istream.read(struct.calcsize(fmt))
+                )
+                if 3 <= datfile_version < 6:
+                    fmt = "=ii"
+                    offset_tree, offset_blocks = struct.unpack(
                         fmt, istream.read(struct.calcsize(fmt))
                     )
-                    if 3 <= datfile_version < 6:
-                        fmt = "=ii"
-                        offset_tree, offset_blocks = struct.unpack(
-                            fmt, istream.read(struct.calcsize(fmt))
-                        )
-                        istream.seek(0, 2)
-                        file_size = istream.tell()
-                        validation = (
-                            offset_tree < file_size and offset_blocks < file_size
-                        )
-            except Exception:
-                pass
-        return validation
+                    istream.seek(0, 2)
+                    file_size = istream.tell()
+                    return offset_tree < file_size and offset_blocks < file_size
+                elif datfile_version < 15:
+                    mylog.warning(
+                        "This may be a amrvac output, "
+                        f"but this format version ({datfile_version}) is not supported yet."
+                    )
```

Comment:
This is brand new but as the primary author of the function I'm merely fixing a case I forgot to factor in.